### PR TITLE
Migrate MUI `Card` to `DashboardCard`

### PR DESCRIPTION
**Pull request by Cloud11PL**
## What type of PR is this?
- [ ] 💅 Refactor
- [ ] 🌟 Feature
- [ ] 🔥 Bug Fix
- [ ] 🔩 Maintenance
- [ ] 🛠 Workflow CI/CD changes

## Related Issues or Documents

- closes #

## Have you written tests?
- [ ] Yes!
- [ ] No... here is why: _Writing tests are mandatory, please replace this text with why test are not included in this PR_

### CODE REVIEW

File: src/attributes/components/AttributeOrganization/AttributeOrganization.tsx [hunk 1]
```diff
@@ -73,12 +75,12 @@ const AttributeOrganization: React.FC<AttributeOrganizationProps> = props => {
             label={
               <>
                 <FormattedMessage id="v1pNHW" defaultMessage="Attribute Class" />
-                <Typography variant="caption">
+                <Text fontWeight="light">
```
Comment (poulch) [inline src/attributes/components/AttributeOrganization/AttributeOrganization.tsx:78]: question: `variant="caption"` has different font color, is there any reason not to apply it here?

File: src/attributes/components/AttributeProperties/AttributeProperties.tsx [hunk 2]
```diff
@@ -122,17 +124,17 @@ const AttributeProperties: React.FC<AttributePropertiesProps> = ({
           label={
             <>
               <FormattedMessage {...messages.visibleInStorefront} />
-              <Typography variant="caption">
+              <Text fontWeight="light">
```
Comment (poulch) [inline src/attributes/components/AttributeProperties/AttributeProperties.tsx:127]: question: same here?

File: src/channels/components/ChannelAllocationStrategy/ChannelAllocationStrategy.tsx [hunk 3]
```diff
@@ -77,13 +76,13 @@ const ChannelAllocationStrategy: React.FC<ChannelAllocationStrategyProps> = ({
...
                 data-test-id={`channel-allocation-strategy-option-${option.type}`}
               >
-                <Typography variant="body1">
+                <Text>
                   <FormattedMessage {...option.title} />
-                </Typography>
+                </Text>
                 {option.subtitle && (
-                  <Typography color="textSecondary" variant="caption">
+                  <Text fontWeight="light">
```
Comment (poulch) [inline src/channels/components/ChannelAllocationStrategy/ChannelAllocationStrategy.tsx:83]: suggestion: missing color

File: src/components/Alert/InlineAlert.tsx [hunk 4]
```diff
@@ -19,7 +20,7 @@ interface AlertCardProps {
 const AlertCard: React.FC<AlertCardProps> = ({ children }) => {
   const classes = useStyles({});
 
-  return <Card className={classes.container}>{children}</Card>;
+  return <DashboardCard className={classes.container}>{children}</DashboardCard>;
```
Comment (poulch) [inline src/components/Alert/InlineAlert.tsx:23]: question: Do you need this extra class? Couldn't you migrate those styles if you need them?

File: src/components/Card/Title.tsx [hunk 5]
```diff
@@ -3,12 +3,29 @@ import React from "react";
 
 interface TitleProps {
   paddingX?: Sprinkles["paddingX"];
+  dataTestId?: string;
+  // "title" is deprecated, use "children" instead
+  title?: React.ReactNode;
+  subtitle?: React.ReactNode;
```
Comment (poulch) [inline src/components/Card/Title.tsx:8]: suggestion: Title should be only wrapper, imo adding this kind of prop makes it less flexible 
You can remove this props and use Title like this:
```
<Title>
    Discounts
     <Text size={4} fontWeight="light">
        3 discounts
      </Text>
</Title>
```

File: src/components/Card/Toolbar.tsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { Box, Sprinkles } from "@saleor/macaw-ui-next";
+import React from "react";
+
+export const Toolbar: React.FC<Sprinkles> = ({ children }) => (
+  <Box position="absolute" right={6} top={6} display="flex" flexDirection="row" gap={2}>
+    {children}
+  </Box>
+);
```
Comment (poulch) [inline src/components/Card/Toolbar.tsx:8]: question: Is it possible that Toolbar cover Title or other content?

**Review by poulch — COMMENTED** [review]
Try to eleminate className with styles from mkeStyles, in most cases you don't need this styles

File: src/components/Card/Toolbar.tsx (hunk 6 above)
Comment (Cloud11PL) [reply src/components/Card/Toolbar.tsx]: I changed it so that it won't

File: src/components/Card/Title.tsx (hunk 5 above)
Comment (Cloud11PL) [reply src/components/Card/Title.tsx]: I made some changes regarding this 😺

File: src/components/Card/Content.tsx [hunk 7]
```diff
@@ -1,7 +1,9 @@
-import { Box, Sprinkles } from "@saleor/macaw-ui-next";
+import { Box } from "@saleor/macaw-ui-next";
 import React from "react";
 
-export const Content: React.FC<Sprinkles> = ({ children, ...rest }) => (
+type ContentProps = React.ComponentProps<typeof Box>;
+
+export const Content: React.FC<ContentProps & { className?: string }> = ({ children, ...rest }) => (
```
Comment (poulch) [inline src/components/Card/Content.tsx:6]: suggestion: You could use type PropsWithBox

File: src/components/Card/Root.tsx [hunk 8]
```diff
@@ -1,8 +1,11 @@
 import { Box, Sprinkles } from "@saleor/macaw-ui-next";
 import React from "react";
 
-export const Root: React.FC<Sprinkles> = ({ children, ...rest }) => (
-  <Box display="flex" flexDirection="column" gap={5} {...rest}>
+export const Root: React.FC<Sprinkles & { className?: string; onClick?: () => void }> = ({
```
Comment (poulch) [inline src/components/Card/Root.tsx:4]: suggestion: You could use type PropsWithBox

File: src/components/Card/Subtitle.tsx [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { Sprinkles, Text } from "@saleor/macaw-ui-next";
+import React, { PropsWithChildren } from "react";
+
+type CardSubtitleProps = Sprinkles;
+
+export const CardSubtitle: React.FC<PropsWithChildren<CardSubtitleProps>> = ({ children }) =>
+  children ? (
+    <Text size={4} fontWeight="light" display="block">
```
Comment (poulch) [inline src/components/Card/Subtitle.tsx:8]: question: what about passing rest of Text props, for example to move text if needed?

File: src/components/Card/Title.tsx [hunk 10]
```diff
@@ -1,14 +1,48 @@
 import { Box, Sprinkles, Text } from "@saleor/macaw-ui-next";
 import React from "react";
 
-interface TitleProps {
-  paddingX?: Sprinkles["paddingX"];
+interface TitleProps extends Sprinkles {
```
Comment (poulch) [inline src/components/Card/Title.tsx:4]: suggestion: You could use type PropsWithBox

File: src/components/Card/Actions.tsx [hunk 11]
```diff
@@ -0,0 +1,24 @@
+import { Box, Sprinkles } from "@saleor/macaw-ui-next";
+import React from "react";
+
+export const Actions: React.FC<Sprinkles & { className?: string; style?: React.CSSProperties }> = ({
```
Comment (poulch) [inline src/components/Card/Actions.tsx:4]: suggestion: You could use type PropsWithBox

File: src/categories/components/CategoryProducts/CategoryProducts.tsx [hunk 12]
```diff
@@ -28,34 +28,36 @@ export const CategoryProducts = ({
...
-    <DashboardCard.Title>
-      <Box display="flex" justifyContent="space-between" alignItems="center">
-        <FormattedMessage
-          id="+43JV5"
-          defaultMessage="Products in {categoryName}"
-          description="header"
-          values={{ categoryName: category?.name }}
-        />
+    <DashboardCard.Header>
+      <DashboardCard.Title>
```
Comment (poulch) [inline src/categories/components/CategoryProducts/CategoryProducts.tsx:32]: suggestion: `DashboardCard.Title` should contains only title, rest like the buttons should be in `DashboardCard.Header`

File: src/components/Card/Actions.tsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
+import { Box, PropsWithBox } from "@saleor/macaw-ui-next";
+import React from "react";
+
+export const Actions: React.FC<PropsWithBox<{ children: React.ReactNode }>> = ({
```
Comment (poulch) [inline src/components/Card/Actions.tsx:4]: question: Do you need this, seems to be not used nowhere

File: src/components/Card/Toolbar.tsx [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  children,
+  ...rest
+}) => <Box {...rest}>{children}</Box>;
+
+export const toolbarWrapperStyles = {
+  display: "flex",
+  flexDirection: "row",
+  alignItems: "center",
+  justifyContent: "space-between",
+} as const;
```
Comment (poulch) [inline src/components/Card/Toolbar.tsx:14]: question: Can you explain me why? :D You should pass this props to Box inside Toolbar component

Comment (poulch) [reply src/components/Card/Toolbar.tsx:14]: Alternatively, you could add props to DashboardCard.Header for example `withToolbar` and then apply those styles or those styles should be used there by default

File: src/components/Card/Header.tsx [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import { Box, PropsWithBox } from "@saleor/macaw-ui-next";
+import React from "react";
+
+export const toolbarWrapperStyles = {
+  display: "flex",
+  flexDirection: "row",
+  alignItems: "center",
+  justifyContent: "space-between",
+} as const;
```
Comment (poulch) [inline src/components/Card/Header.tsx:9]: suggestion: I'd add more strict typing, albo do we need to export this styles?

File: src/components/Card/index.ts [hunk 16]
```diff
@@ -1,5 +1,16 @@
+import { Actions as BottomActions } from "./Actions";
```
Comment (poulch) [inline src/components/Card/index.ts:1]: suggestion: I'd suggest leave Action, as Action can be place anywhere but `BottomActions` implies something wrong

File: src/components/Card/Toolbar.tsx [hunk 17]
```diff
@@ -0,0 +1,7 @@
+import { Box, PropsWithBox } from "@saleor/macaw-ui-next";
+import React from "react";
+
+export const Toolbar: React.FC<PropsWithBox<{ children: React.ReactNode }>> = ({
```
Comment (poulch) [inline src/components/Card/Toolbar.tsx:4]: suggestion: `Toolbar` does nothing it's only play wrapper role, do you need them?

**Review by poulch — APPROVED** [review]

**Review by poulch — DISMISSED** [review]

File: src/components/Card/index.ts (hunk 16 above)
Comment (Cloud11PL) [reply src/components/Card/index.ts:1]: Does it really imply something's wrong? My intention was to show where the actions should be placed and to differentiate them from Toolbar.

**Review by poulch — APPROVED** [review]

**Review by yellowee — APPROVED** [review]

**Pull request merged**